sdv/single_table/base.py

```diff
@@ -1335,14 +1336,29 @@ def _sample_with_conditions(
 
             condition = dict(zip(condition_columns, group))
             condition_df = dataframe.iloc[0].to_frame().T
-            try:
-                transformed_condition = self._data_processor.transform(
-                    condition_df, is_condition=True
-                )
-            except ConstraintsNotMetError as error:
-                raise ConstraintsNotMetError(
-                    'Provided conditions are not valid for the given constraints.'
-                ) from error
+            if hasattr(self, '_chained_patterns'):
+                try:
+                    transformed_condition = self._transform_helper(condition_df)
+                    transformed_condition = self._data_processor.transform(
+                        transformed_condition, is_condition=True
+                    )
+                except PatternNotMetError:
+                    raise PatternNotMetError(
+                        'Provided conditions are not valid for the given constraints.'
+                    )
+                except Exception:
```

Comment:
Is it possible to catch a more specific exception?

Reply:
Not really, we basically want to try to transform the conditions using the constraints but since we only have a subset of the columns it's likely to crash with either a key error or indexing error from pandas when we try to access a column that doesn't exist. We're only guaranteed to get a PatternNotMetError if (1) all necessary columns are present for data validation and (2) data validation fails.

Reply:
If it fails to transform it's fine since we just fall back to reject sampling for the constraints.

Reply:
Gotcha